react/src/pages/utils/components/BooleanFilter.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from "react";
+import { FormControl, MenuItem, Select } from "@material-ui/core";
+import { Column } from "material-table";
+import { PseudoBoolean, PseudoBooleanReadableMap } from "../typings";
+
+/**
+ * A controlled component to override material-table columns' editComponent prop.
+ */
+export default function BooleanFilter<RowData extends object>(props: {
+    columnDef: Column<RowData>;
+    onFilterChanged: (columnID: string, value: any) => void;
+}) {
+    const options = Object.keys(PseudoBooleanReadableMap) as PseudoBoolean[];
+    const [value, setValue] = useState<PseudoBoolean | "">("");
+
+    return (
+        <FormControl>
+            <Select
+                fullWidth
+                value={value}
+                onChange={e => {
+                    setValue(e.target.value as PseudoBoolean);
+                    props.onFilterChanged((props.columnDef as any).tableData.id, e.target.value);
+                }}
+            >
+                <MenuItem value="" key="blank">
+                    (Blank)
+                </MenuItem>
+                {options.map(key => (
```

Comment:
Not sure why we have "Not specified" in the filter. It's not one of the options that the user can select so it's a bit confusing.

Reply:
It depends if it's possible for the backend to send `undefined` as a value for these columns. Any editing on the frontend only allows `true`, `false`, and `null`, but we allow filtering for `undefined` as well.

Reply:
So to the users, is there a difference between "not specified" and "unknown"?

Reply:
There is because the default state of the filter is to not filter at all, as opposed to filtering specifically for the unspecified null state.

Reply:
Do you mean "not specified" = no filter applied? I thought the "(blank)" option represents the no filter state.

Reply:
Just to clear things up, these are the possible filter values:
- "(Blank)": no filter applied, show all rows
- "Yes": filter for `true` values
- "No": filter for `false` values
- "Unknown": filter for `null` values
- "Not specified": filter for `undefined` values

As far as semantic differences, "Unknown" indicates that the user filled out the field, and specifically said that they don't know that the field is "Yes" or "No". "Not specified" indicates that the field is not filled out (maybe leftover after migrations from old sample tracker or something).

Reply:
Okay, that makes sense. Thanks for the clarification!

Reply:
A null field in the database is None in Python and should be null in JSON. `undefined` isn't a value in JSON, only JavaScript. I thought you were using `undefined` to indicate there was no filter applied but this actually has one more state than required then 👀

Reply:
Ah gotcha. I can remove it then.